packages/compass-components/src/hooks/use-confirmation.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+
+import {
+  default as ConfirmationModal,
+  Variant as ConfirmationModalVariant,
+} from '@leafygreen-ui/confirmation-modal';
+
+export { ConfirmationModalVariant };
+
+type ConfirmationProperties = {
+  title: string;
+  description: string;
+  buttonText?: string;
+  variant?: ConfirmationModalVariant;
+};
+type ConfirmationCallback = (value: boolean) => void;
+
+interface ConfirmationModalContextData {
+  showConfirmation: (
+    props: ConfirmationProperties,
+    callback: ConfirmationCallback
+  ) => void;
+  isMounted: boolean;
+}
+
+const ConfirmationModalContext =
+  React.createContext<ConfirmationModalContextData>({
+    isMounted: false,
+  } as ConfirmationModalContextData);
+
+class ShowConfirmationEvent extends Event {
+  constructor(
+    public props: ConfirmationProperties,
+    public callback: ConfirmationCallback
+  ) {
+    super('show-confirmation');
+  }
+}
+class GlobalConfirmation extends EventTarget {
+  showConfirmation(props: ConfirmationProperties) {
+    return new Promise<boolean>((resolve) => {
+      const event = new ShowConfirmationEvent(props, resolve);
+      this.dispatchEvent(event);
+    });
+  }
+}
+const globalConfirmation = new GlobalConfirmation();
+export const showConfirmation =
+  globalConfirmation.showConfirmation.bind(globalConfirmation);
+
+type ConfirmationModalAreaProps = ConfirmationProperties & {
+  open: boolean;
+};
+
+export const ConfirmationModalArea: React.FC = ({ children }) => {
+  const hasParentContext = useContext(ConfirmationModalContext).isMounted;
+
+  const [confirmationProps, setConfirmationProps] = useState({
+    open: false,
+  } as ConfirmationModalAreaProps);
```

Comment:
Same here, I am not sure why you are asserting here, your interfaces should match what you are doing